lib/assessment.js

```diff
@@ -4,24 +4,43 @@
 // Write a function that takes zero or more arguments
 // and returns the minimum from that set
 // assume that all the argument are numbers
-var min = function min() {
 
-};
+//arguments = "array-ish"
+var min = function min() {
+  var result = arguments[0]
+  for(var index; index<arguments.length; index ++)
+    if(arguments[index]<result) {
+      result=arguments[index]
+    }
+return result
+}`
+//I used mdn
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
-var sum = function sum() {
 
-};
+var sum = function sum(){
+  var result = 0;
+    for (var index = 0; index < args.length; index ++) {
+      result = result + args[index];
+    }
+  return result;
+}
+
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 var newArray = function newArray(length, defaultValue) {
-
+  var result = [];
+    for (i=0; i < length; i++) {
```

Comment:
`i` is not declared here, but it is assigned; this means the interpreter will go out into the surrounding scope looking for a variable called `i` and overwrite it! This can cause very puzzling problems in your code that are difficult to debug.